src/vulkan-renderer/wrapper/device.cpp

```diff
@@ -326,21 +326,34 @@ Device::Device(const wrapper::Instance &instance, const VkSurfaceKHR surface, bo
 
     replay_file_test.close();
 
-    // VMA allows to record memory allocations to a .csv file.
+    // VMA allows recording memory allocations to a .csv file.
     // This .csv file can be replayed using tools from the repository.
     // This is very useful every time there is a bug in memory management.
     VmaRecordSettings vma_record_settings;
 
     // We flush the stream after every write operation because we are expecting unforeseen program crashes.
-    // This might has a negative effect on the application's performance but it's worth it for now.
+    // This might have a negative effect on the application's performance, but it's worth it for now.
     vma_record_settings.flags = VMA_RECORD_FLUSH_AFTER_CALL_BIT;
     vma_record_settings.pFilePath = vma_replay_file.c_str();
 
     VmaAllocatorCreateInfo vma_allocator_ci{};
     vma_allocator_ci.physicalDevice = m_graphics_card;
     vma_allocator_ci.instance = instance.instance();
     vma_allocator_ci.device = m_device;
-    vma_allocator_ci.vulkanApiVersion = instance.vulkan_api_version();
+
+    const auto vk_api_version = instance.vulkan_api_version();
+
+    // VMA can't handle Vulkan API version newer than 1.1
+    if (VK_VERSION_MAJOR(vk_api_version) > VK_VERSION_MAJOR(VK_API_VERSION_1_1) ||
```

Comment:
This should only compare the major version. Though, I still think we should just hardcode the instance version as 1.2 and the VMA version as 1.1, would remove the need for all this logic, then we could just have the `vkEnumerateInstanceVersion` to make sure 1.2 is available.

Reply:
Inside of VMA both major and minor version are being compared in the assertion though. However I think you are right. Let's just do it the way you described.

Reply:
Ah right yeah, it would be `MAJOR(a) > MAJOR(1_1) || (MAJOR(a) == MAJOR(1_1) && MINOR(a) > MINOR(B))` I think. But not doing it all is better :D